src/core/MultiFormatReader.ts

```diff
@@ -101,41 +101,42 @@ export default class MultiFormatReader implements Reader {
 
         const tryHarder: boolean = hints !== null && hints !== undefined && undefined !== hints.get(DecodeHintType.TRY_HARDER);
         /*@SuppressWarnings("unchecked")*/
-        const formats = hints === null || hints === undefined ? null : hints.get(DecodeHintType.POSSIBLE_FORMATS);
+        const formats = hints === null || hints === undefined ? null : <BarcodeFormat[]>hints.get(DecodeHintType.POSSIBLE_FORMATS);
```

Comment:
I found it interesting that in Java version the hints is typed too: https://github.com/zxing/zxing/blob/0cf3b9be71680f50c90a71ca26ce0d33664b0dd6/core/src/main/java/com/google/zxing/MultiFormatReader.java#L101

Reply:
`hints` in Java is of type `Map<DecodeHintType,?>`
`hints` in Typescript is of type `Map<DecodeHintType, any>`

So, as far as I can see, each one only specifies the Map's key type, but not the maps value type. This is, IMO, because the different `DecodeHintType.XXX` keys use values of different types. So after getting a value `hints.get(DecodeHintType.XXX)` it is necessary to cast to the desired type:

```Java
// Java
(Collection<BarcodeFormat>) hints.get(DecodeHintType.POSSIBLE_FORMATS)
```
```typescript
// Typescript
<BarcodeFormat[]>hints.get(DecodeHintType.POSSIBLE_FORMATS)
```